src/triangular.jl

```diff
@@ -61,3 +61,39 @@ end
 function _first_zero_on_diagonal(A::StaticULT)
     _first_zero_on_diagonal(A.data)
 end
+
+
+inv(R::UpperTriangular{T, <:StaticMatrix}) where T =
+    (@inline; UpperTriangular(_inv_upper_triangular(R.data)))
+
+_inv_upper_triangular(R::StaticMatrix{n, m}) where {n, m} =
+    checksquare(R)
+
+@generated function _inv_upper_triangular(R::StaticMatrix{n, n, T}) where {n, T}
+    ex = quote
+        R_inv = MMatrix{n,n,T}(undef)
```

Comment:
You should either use `similar` (to properly handle non-isbits `T`) or the same trick as `_A_ldiv_B`. Also, make sure the method works to `T` being an integer.